src/globus_cli/commands/collection/create/guest.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import annotations
+
+import typing as t
+import uuid
+
+import click
+import globus_sdk
+import globus_sdk.experimental.auth_requirements_error
+
+from globus_cli.commands.collection._common import (
+    filter_fields,
+    standard_collection_fields,
+)
+from globus_cli.constants import ExplicitNullType
+from globus_cli.endpointish import EntityType
+from globus_cli.login_manager import (
+    LoginManager,
+    MissingLoginError,
+    read_well_known_config,
+)
+from globus_cli.login_manager.context import LoginContext
+from globus_cli.parsing import command, endpointish_params, mutex_option_group
+from globus_cli.services.gcs import CustomGCSClient
+from globus_cli.termio import TextMode, display
+
+
+@command("guest", short_help="Create a GCSv5 Guest Collection")
+@click.argument("mapped-collection-id", type=click.UUID)
+# TODO: Update this to guest-collection-base-path
+@click.argument("collection-base-path", type=str)
```

Comment:
I'm a little bit confused by this TODO item.
Why would we call it `guest_collection_base_path`?

Also, a minor matter of conventions: I have a mild preference for using `"COLLECTION_BASE_PATH"` as the param name because it matches the metavar used in the usage text (likewise, `"MAPPED_COLLECTION_ID"`). I should maybe create a lint for this to ensure that we case and separate our `click.argument`s consistently.

Reply:
Oh whoops. This was a wording note I left for myself while reviewing the command with Brigitte.

Can definitely update the wording to be uppercase.

Reply:
Stephen mentioned he will discuss `GUEST_COLLECTION_BASE_PATH` vs `COLLECTION_BASE_PATH` w/ Brigitte separately & get back to me.

Reply:
We just finished up that conversation (such response times, wow! :dash: ), and the upshot is this:
- we'll keep the name as `COLLECTION_BASE_PATH` to match the API
- the helptext should clarify the meaning of the three positional arguments

I very quickly dashed off this bit of text, which we agreed needs a little bit more wordsmithing:
```
  Create a GCSv5 Guest Collection.

  Requires the ID of the Mapped Collection which hosts the Guest Collection,
  as well as a path on the Mapped Collection and a name for the new Guest
  Collection.
```

As an alternative, just to seed some ideas, it could also be
```
  Create a GCSv5 Guest Collection.

  Requires the host collection ID, the path under that collection, and a name.
```

Helptext should be easy to expand in the docstring (below).

I'm not too attached to any particular wording; the only requirement here is that we clarify that the `COLLECTION_BASE_PATH` refers to the location on the mapped collection where the guest lives. The concern we're addressing is the potential for a user to believe that they are expected to supply the base path of the Mapped Collection on the Storage Gateway.

Reply:
Cool! I have some feedback on the specific help-text but let's close this thread as resolved and treat details of the help as new discussions. :+1: